etc/otel-jmx.config.yaml

```diff
@@ -12,25 +11,46 @@ rules:
        metric: kafka.streams.records.lag
        type: gauge
        desc: the current lag of the partition
-       unit: '{messages}'
+       unit: '{records}'
  - bean: kafka.consumer:type=consumer-fetch-manager-metrics,client-id=*
    metricAttribute:
      clientid: param(client-id)
    mapping:
      records-lag-max:
        metric: kafka.streams.records.lag.max
        type: gauge
-       desc: the current lag of the partition
-       unit: '{messages}'
- - bean: kafka.streams:type=stream-thread-metrics,thread-id=*
+       desc: the max lag of all partitions
+       unit: '{records}'
+ - bean: kafka.consumer:type=consumer-coordinator-metrics,client-id=*
    metricAttribute:
-     thread: param(thread-id)
+     clientid: param(client-id)
    mapping:
-     process-total:
-       metric: kafka.streams.thread.process.total
+     rebalance-rate-per-hour:
+       metric: kafka.streams.rebalance.rate
        type: gauge
-       desc: total records processed
-       unit: '{records}'
+       desc: the rate of rebalances
+       unit: '{rebalances-per-hour}'
+     failed-rebalance-rate-per-hour:
+       metric: kafka.streams.failed.rebalance.rate
+       type: gauge
+       desc: the rate of failed rebalances
+       unit: '{rebalances-per-hour}'
+
+ # Kafka Streams application metrics
+ ## CAUTION: these are referred to as "client metrics" in the Streams docs, but we call
+ ## them "application metrics" to avoid confusion with the consumer/producer clients.
+ ## Note the translation from 'client-id' to 'application-id' in our tags
+ - bean: kafka.streams:type=stream-metrics,client-id=*
+   metricAttribute:
+     application-id: param(client-id)
+   mapping:
+     state:
+       metric: kafka.streams.state
+       type: gauge
+       desc: the current state of this Streams application
+       unit: '{KafkaStreams.State}'
```

Comment:
Any opinion on whether to give this a unit at all?

Reply:
I don't think it matters.